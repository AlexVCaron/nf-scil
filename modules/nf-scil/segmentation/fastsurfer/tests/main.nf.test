nextflow_process {

    name "Test Process SEGMENTATION_FASTSURFER"
    script "../main.nf"
    process "SEGMENTATION_FASTSURFER"

    tag "modules"
    tag "modules_nfcore"
    tag "segmentation"
    tag "segmentation/fastsurfer"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "Freesurfer.zip",
                                               "T1w.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("segmentation - fastsurfer") {
        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        freesurfer: it.simpleName == "Freesurfer"
                        anat: it.simpleName == "T1w"
                    }
                ch_license = ch_split_test_data.freesurfer.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/license.txt")
                    ]
                }
                ch_anat = ch_split_test_data.anat.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/T1w.nii.gz")
                    ]
                }
                input[0] = ch_anat
                    .join(ch_license)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.fastsurferdirectory.get(0).get(1)).name,
                    process.out.versions).match()}
            )
        }
   }
}
