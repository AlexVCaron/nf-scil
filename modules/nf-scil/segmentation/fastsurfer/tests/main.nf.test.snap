{
    "segmentation - fastsurfer": {
        "content": [
            "test_fastsurfer"
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-04T22:07:50.070648"
    }
}